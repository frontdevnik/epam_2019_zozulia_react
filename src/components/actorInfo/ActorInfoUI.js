import React from 'react';
import PropTypes from 'prop-types';
import withTranslation from '../../hocs/withTranslation';
import styles from './actorInfo.module.scss';

function ActorInfoUI(props) {
  const { name, imgUrl, biography } = props.actor;
  const { actor_name, actor_biography } = props;

  return (
    <div className={styles.aboutActor}>
      <img className={styles.profileImg} alt={name} src={imgUrl} />
      <div>
        <p className={styles.actorInfo}>
          <b>{actor_name}: </b>
          {name}
        </p>
        <p className={styles.actorInfo}>
          <b>{actor_biography}: </b>
          {biography}
        </p>
      </div>
    </div>
  )
};

ActorInfoUI.propTypes = {
  actor: PropTypes.shape({
    name: PropTypes.string,
    imgUrl: PropTypes.string,
    biography: PropTypes.string,
    actor_name: PropTypes.string,
    actor_biography: PropTypes.string,
  })
};

const withTranslationWorlds = withTranslation(['actor_name', 'actor_biography']);

export default withTranslationWorlds(ActorInfoUI);
