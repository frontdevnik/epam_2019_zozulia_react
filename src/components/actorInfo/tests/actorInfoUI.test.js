import React from 'react';
import {TestActorInfoUI} from "../ActorInfoUI";
import {actorInfoProps} from "./stabProps";
import styles from "../actorInfo.module.scss";

const setUp = (props) => shallow(<TestActorInfoUI {...props} />);

describe('<ActorInfoUI />', () => {
    let component;

    beforeEach(() => {
        component = setUp(actorInfoProps);
    })

    it('should render correctly', () => {
        expect(component.html()).toMatchSnapshot()
    })
    it('should render img correctly', () => {
        expect(component.find('img').prop('src')).toBe(actorInfoProps.actor.imgUrl);
        expect(component.find('img').prop('alt')).toBe(actorInfoProps.actor.name);
    })
    it('should render actor name correctly', () => {
        const node = component.find(`.${styles.actorInfo}`);
        expect(node.first().text()).toBe(`${actorInfoProps.actor_name}: ${actorInfoProps.actor.name}`);
    })
    it('should render actor biography correctly', () => {
        const node = component.find(`.${styles.actorInfo}`);
        expect(node.last().text()).toBe(`${actorInfoProps.actor_biography}: ${actorInfoProps.actor.biography}`);
    })
});
