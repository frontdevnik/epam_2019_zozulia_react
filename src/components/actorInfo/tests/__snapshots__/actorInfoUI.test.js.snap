// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActorInfoUI /> should render correctly 1`] = `"<div class=\\"aboutActor\\"><img class=\\"profileImg\\" alt=\\"name\\" src=\\"img\\"/><div><p class=\\"actorInfo\\"><b>Name: </b>name</p><p class=\\"actorInfo\\"><b>Biography: </b>biography</p></div></div>"`;
